### ISSUE TITLE
Difficulty in Training LiteFlowNet

### ISSUE THREAD

**Issue by syed-mujtaba-hassan** [twhui/LiteFlowNet#23]
Hi, 

I am trying to train model for LiteFlowNet using the methodology mentioned in the paper.  However, I am facing difficulties in running inference on the partial trained networks. Correct me if I am wrong. The method that I understood is to train till sub pixel refinement part of layer 6 first, followed by regularization part of layer 6 and then continue the same procedure for other layers.  However, after training this way, when I run inference on the sub-portion of complete network (till the part where training has been done), the inference takes too much time outputting something like "Batch 0, img0/1_aug_L2/3/4/5 = some number" repeatedly. However, when I run inference on the whole network, inference is fast and this repetition does not take place. Can you guide me, what I am doing wrong.

Thanks for reading the post.

**Comment by twhui** [comment]
Hi,

The training procedure is mentioned in the 2nd paragraph of Sect. 4. I just want to point out that the stage-wise training scheme is used only for pre-training on Chairs dataset (i.e. those models with "-pre"). 

For pre-training, say L6, you need to remove the code segment for L5 to L2. Similarly, pre-training upto L5, you need to remove the code segment for L4 to L2.

**Comment by syed-mujtaba-hassan** [comment]
Thanks for the quick reply and clarifying the procedure of training.

However, I guess I was not clear in asking my question. Actually, I want to ask that when I use this partial trained network, and run inference on that part of network (i.e till layer 6 + layer 1 which does not require any training) to calculate the flow, the inference takes too much time giving the outputs as stated in the earlier post. However, when I run inference on the whole network (either using the trained model provided in your github repository or my partially trained model which is trained only till layer 6), the inference is fast and no repetitions of the output takes place. However, using whole network for inference, while training is still going on does not seem logical to me as the weights of other layers (i.e layer 2) are not trained. I want to know what I am doing wrong and how can I check the intermediate flow results. Is it something related to data augmentation / crop size causing multiple images to be generated during inference time that I am missing?

Secondly, I get slightly confused regarding the training procedure. Please correct me if I am wrong. The procedure for training is : 
1). LiteFlowNet-pre: Train the network stage-wise as mentioned in your answer using FlyingChairs dataset
2). LiteFlowNet: Using the trained LiteFlownet-pre model, fine-tune this complete model (no training in stages needs to be done for this fine-tuning) on FlyingThings3D dataset (In this case, can you tell what learning rate you are using as I cannot find this out in the paper). 

Thanks once again for reading the post. I am sorry if my questions seems irrelevant / silly.

**Comment by twhui** [comment]
Generally speaking, support other than issue related to this github is not provided! Anyway, I have no idea on what you meant by "till layer 6 + layer 1"? If you only want to infer flow fields say level 6, then you need to comment out all unused layers (i.e. from levels 5 to 2). 

The procedure is correct.

**Comment by syed-mujtaba-hassan** [comment]
Thank you for the response and giving your time. 

From layer 6 + layer 1, I meant that for inference, I used the network till layer 6 and at the end give the output of layer 6 as input to layer 1 (post processing layer). In layer 1, there is no trainable parameters for the convolution layer, "ScaleMag_flow". So, I think that I can use this post processing layer and added it for inference as this layer also write flow to a file which I require for evaluating performance of model.  However, I left the weight and height parameters in the post processing layer as it is ( width: $TARGET_WIDTH, height: $TARGET_HEIGHT) and am not sure whether this is correct or not.

I am sorry if the questions are not related to this github.
